### ISSUE TITLE
Preview has become a bit stale

### ISSUE THREAD

**Issue by KristofferC** [JuliaLang/Pkg.jl#591]
Not sure how useful the feature is. Should be removed or checked that it doesn't modify any state (updating registry should be ok though).

**Comment by fredrikekre** [comment]
I propose that, if we keep `preview`, then everything works as normal with `preview`, except that writing to `Project.toml`/`Manifest.toml` is blocked. For example, we don't currently update the registries with `preview`, and therefore the `preview` status of a `add` call, for example, might not match what you get when you call `add` without preview.

**Comment by StefanKarpinski** [comment]
We could also just save a couple of different older states of project and manifest files even if they're not checked into git so that we can undo changes instead. Saving some small text files would be quite cheap.

**Comment by KristofferC** [comment]
It's not the Project + Manifest that is the problem, writing these are already guarded when `preview` is active so that will never happen.

**Comment by StefanKarpinski** [comment]
Yes, but if we just always saved those then we could drop the preview feature entirely since the main point of it seems to be seeing if you want to do something or not. Instead you can just do it and then undo it if you didn't like what happened.

**Comment by simonbyrne** [comment]
It's not working for me:
```
(v1.0) pkg> preview add StaticArrays
ERROR: Wrong number of arguments
```

**Comment by simonbyrne** [comment]
Nevermind, that was fixed in the latest version.
